### PR TITLE
Add troubleshooting guide for VMs

### PR DESCRIPTION
**Pull request by howardjohn**
This was requested as part of the VM graduation support ability review

### CODE REVIEW

File: content/en/docs/ops/diagnostic-tools/virtual-machines/index.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+$ istioctl proxy-status
+NAME           CDS        LDS        EDS        RDS      ISTIOD                    VERSION
+vm-1.default   SYNCED     SYNCED     SYNCED     SYNCED   istiod-789ffff8-f2fkt     {{< istio_full_version >}}
+{{< /text  >}}
+
+However, `istioctl proxy-config` relies on functionality in Kubernetes to connect to a proxy, which will not work for virtual machines.
+Instead, a file containing the configuration dump from Envoy can be passed. For example:
+
+{{< text bash >}}
+$ curl -s localhost:15000/config_dump | istioctl proxy-config clusters --file -
```
Comment (jasonwzm) [inline content/en/docs/ops/diagnostic-tools/virtual-machines/index.md:80]: There should be a step before this to move istioctl to the VM, correct?

File: content/en/docs/ops/diagnostic-tools/virtual-machines/index.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+the virtual machine to become a part of a `Service`, similar to an `Endpoint` in Kubernetes.
+
+To check these are created correctly:
+
+{{< text bash >}}
+$ kubectl get workloadentries
+NAME             AGE   ADDRESS
+vm-10.128.0.50   14m   10.128.0.50
+{{< /text  >}}
+
```
Comment (jasonwzm) [inline content/en/docs/ops/diagnostic-tools/virtual-machines/index.md:100]: I think we should add some details on troubleshooting workload entries not created automatically. grepping the istiod logs may be good enough for now.

File: content/en/docs/ops/diagnostic-tools/virtual-machines/index.md (hunk 1 above)
Comment (howardjohn) [reply content/en/docs/ops/diagnostic-tools/virtual-machines/index.md:80]: I don't think users should actually put istioctl on the VM. The should extract a config dump through various means specific to their environment and then run istioctl somewhere else. We cannot really easily document how to run commands on a VM - all deployments are bespoke. Instead, we just explain how they can be used

File: content/en/docs/ops/diagnostic-tools/virtual-machines/index.md (hunk 2 above)
Comment (howardjohn) [reply content/en/docs/ops/diagnostic-tools/virtual-machines/index.md:100]: SGTM, I will add a note to look at logs. I also filed a bug to add metrics which can be helpful

File: content/en/docs/ops/diagnostic-tools/virtual-machines/index.md (hunk 1 above)
Comment (stevenctl) [reply content/en/docs/ops/diagnostic-tools/virtual-machines/index.md:80]: What about `ssh user@vm 'curl localhost:15000/config_dump' | istioctl ...` to make it more clear?

**Review by stevenctl — APPROVED** [review]

File: content/en/docs/ops/diagnostic-tools/virtual-machines/index.md (hunk 1 above)
Comment (jasonwzm) [reply content/en/docs/ops/diagnostic-tools/virtual-machines/index.md:80]: Yea, just some way to let the user know the config dump needs to be pulled from the VM.

File: content/en/docs/ops/diagnostic-tools/virtual-machines/index.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+title: Debugging Virtual Machines
+description: Describes tools and techniques to diagnose issues with Virtual Machines.
+weight: 20
+keywords: [debug,virtual-machines,envoy]
+owner: istio/wg-environments-maintainers
+test: n/a
+---
+
+This page describes how to troubleshoot issues with Istio deployed to Virtual Machines.
+Before reading this, you should the steps in [Virtual Machine Installation](/docs/setup/install/virtual-machine/).
```
Comment (rootsongjc) [inline content/en/docs/ops/diagnostic-tools/virtual-machines/index.md:11]:
```suggestion
Before reading this, you should take the steps in [Virtual Machine Installation](/docs/setup/install/virtual-machine/).
```

**Review by nmittler — APPROVED** [review]

**Pull request merged**